packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -233,15 +233,16 @@ class AttachCommand extends FlutterCommand {
           rethrow;
         }
       } else if ((device is IOSDevice) || (device is IOSSimulator)) {
-        observatoryUri = Stream<Uri>
-          .fromFuture(
-            MDnsObservatoryDiscovery.instance.getObservatoryUri(
-              appId,
-              device,
-              usesIpv6: usesIpv6,
-              deviceVmservicePort: deviceVmservicePort,
-            )
-          ).asBroadcastStream();
+        final Uri uriFromMdns =
+          await MDnsObservatoryDiscovery.instance.getObservatoryUri(
+            appId,
+            device,
+            usesIpv6: usesIpv6,
+            deviceVmservicePort: deviceVmservicePort,
+          );
+        observatoryUri = uriFromMdns == null
```

Comment:
Instead of settings this to null with a ternary, set it conditionally:

```
if (uriFromMdns != null) {
  observatoryUri = Stream<Uri>.value(uriFromMdns).asBroadcastStream();
}
```

Reply:
I think being explicit about this value being null at this point is perhaps more readable than assuming it’s already null at this point. Wdyt?

Reply:
SGTM